### ISSUE TITLE
Remove deprecated functions from t.web.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7828]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7828|
|Type|enhancement|
|Created|2015-03-23 08:41:41Z|
|Branch|https://github.com/twisted/twisted/tree/twutil-rmdeprecations-7828|

Since I need to port this module, I noticed there was a bunch of stuff that's been deprecated since 2012 and is awful stuff that nobody should use.

Let's remove it!

<details><summary>Searchable metadata</summary>

```
trac-id__7828 7828
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_twutil_rmdeprecations_7828 branches/twutil-rmdeprecations-7828
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1427100101609514 1427100101609514
changetime__1427280180691527 1427280180691527
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44208]) Branching to twutil-rmdeprecations-7828.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Removed them, pushed up, builders spun.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_3"></a>|@derwolfe set owner to @hawkowl|
|-|-|

Thanks for your work!

1. It looks like this complies with the Twisted deprecation policy detailed at http://twistedmatrix.com/trac/wiki/CompatibilityPolicy#IncompatibleChanges. 
   a. Your patch removes deprecated code three major versions after the deprecation warnings were added (added in 12.1). The policy requires that the warnings be present for at least 2 releases, which they has been.
   b. Your patch removes deprecated code ~2.5/3 years after deprecation (12.1's release date was June 4, 2012). The policy requires that the deprecated code be present with warnings for at least one year, which it certainly has been.
2. Code:
   a. All of the builders are green.
   b. You've done a bit of extra cleanup by adding the ``` __all__ ``` to  ``` twisted.web.util```, nice.
   c. It looks like there are no instances of the deprecated functions/variables lurking around elsewhere in the source. I verified this using ``` git grep ``` on the following:
```
htmlFunc
htmlInst
htmlReprTypes
htmlList
htmlDict
htmlUnknown
saferepr
htmlrepr
```
I was unable to find any instances of ``` twisted.web.util.stylesheet```, but it could have gotten past my search.

The patch looks good to me. Please land it if you feel my review is sufficient!

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Hi herrwolfe!

Thanks for the review!

The __all__ was actually at the top, I just moved it to the bottom while I was changing it :)

Will merge!

-Hawkie

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44259]) Merging twutil-rmdeprecations-7828: Remove deprecated functions from t.web.util

Author: hawkowl
Reviewer: herrwolfe
Fixes: [#7828](https://github.com/twisted/twisted/issues/7828)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Why did you move `__all__`?
